Firestore/Source/Model/FSTMutation.mm

```diff
@@ -443,27 +457,29 @@ - (nullable FSTMaybeDocument *)applyTo:(nullable FSTMaybeDocument *)maybeDoc
 }
 
 /**
- * Transforms the provided `previousValue` via the provided `fieldTransform` which must be an
- * ArrayUnion or ArrayRemove transform. Used both for local application and after server
- * acknowledgement.
+ * Transforms the provided `previousValue` via the provided `elements`. Used both for local
+ * application and after server acknowledgement.
  */
-- (FSTFieldValue *)arrayTransformResultWithPreviousValue:(FSTFieldValue *)previousValue
-                                               transform:(const FieldTransform &)fieldTransform {
-  NSMutableArray *result = [self coercedFieldValuesArray:previousValue];
-  auto array_transform = static_cast<const ArrayTransform &>(fieldTransform.transformation());
-  if (fieldTransform.transformation().type() == TransformOperation::Type::ArrayUnion) {
-    for (FSTFieldValue *element : array_transform.elements()) {
-      if (![result containsObject:element]) {
-        [result addObject:element];
-      }
-    }
-  } else if (fieldTransform.transformation().type() == TransformOperation::Type::ArrayRemove) {
-    for (FSTFieldValue *element : array_transform.elements()) {
-      [result removeObject:element];
+- (FSTFieldValue *)arrayUnionResultWithElements:(const std::vector<FSTFieldValue *>)elements
```

Comment:
You're missing a reference on the elements parameter (should be `const std::vector<FSTFieldValue *> &`), though see above about making this take `const FieldTransform &` instead.

Reply:
Oops, good catch, done.